lib/maths/CDataFrameUtils.cc

```diff
@@ -34,6 +34,27 @@ using TRowItr = core::CDataFrame::TRowItr;
 using TRowRef = core::CDataFrame::TRowRef;
 using TRowSampler = CSampling::CRandomStreamSampler<TRowRef>;
 
+//! Reduce the results of core::CDataFrame::readRows.
+template<typename READER, typename REDUCER, typename FIRST_REDUCER>
+bool doReduce(std::pair<std::vector<READER>, bool> readResults,
+              FIRST_REDUCER reduceFirst,
+              REDUCER reduce) {
```

Comment:
I agree it is better as a binary function. I'm not so sure about calling it a `reduceOperator`, because in a CS context this always makes me think of operators rather functions. I've documented the type requirements instead so it is clear. See [3c43d1c](https://github.com/elastic/ml-cpp/pull/545/commits/3c43d1c0b1bf784a11a15d52e2acaeba54d418d3).